src/java_bytecode/java_types.h

```diff
@@ -110,6 +110,12 @@ class java_generic_parametert:public reference_typet
     return type_variables().front();
   }
 
+  type_variablet &type_variable_ref() const
+  {
+    PRECONDITION(!type_variables().empty());
+    return const_cast<type_variablet &>(type_variables().front());
```

Comment:
I think this is wrong, and could create a very nasty bug in the future. It's a method marked `const`, which means that it guarantees that it won't mutate the internal state of the object, but what it does is it fetches an object from a member container, and removes `const` from that, meaning that you can mutate the object this returns, and therefore mutate the internal state of the object this method is being called at, violating the `const` guarantee of this method.

I might be wrong though.

Reply:
Good catch, `const` removed.